### DISCUSSION TITLE
drizzle plugin: PothosSchemaError: User has not been implemented

### DISCUSSION THREAD

**Discussion by m1212e** [hayes/pothos#1363] [Q&A] [answered]
When using the drizzle plugin I get `PothosSchemaError: User has not been implemented` when defining any `drizzleObject`. This is my schema:

```ts
export const users = pgTable("users", {
	id: serial("id").primaryKey(),
	name: text("name").notNull(),
});

export const usersRelations = relations(users, ({ one, many }) => ({
	posts: many(posts),
}));

export const posts = pgTable("posts", {
	id: serial("id").primaryKey(),
	content: text("content").notNull(),
	authorId: integer("author_id").references(() => users.id),
});

export const postsRelations = relations(posts, ({ one, many }) => ({
	author: one(users, { fields: [posts.authorId], references: [users.id] }),
}));
```
and this is what I do:
```ts
const UserRef = schemaBuilder.drizzleObject("users", {
  name: "User",
  fields: (t) => ({
    id: t.exposeInt("id"),
    name: t.exposeString("name"),
  }),
});
```
do you have any hint at what I'm doing wrong? Thank you so much!

**Comment by hayes** [comment]
Are you importing the file that defines user?

**Reply by hayes** [reply, marked as answer]
This error happens when a field references a type that hasn't been implemented.  If you are implementing the type (which it looks like you are) I would double check you are importing the file where it's implemented before calling builder.toSchema

**Reply by m1212e** [reply]
The `toSchema` hint was it, thank you very much. I accidentially triggered a schema build before finishing registration. That caused it!
